ra/registration-authority.go

```diff
@@ -88,14 +88,21 @@ func validateEmail(address string, resolver core.DNSResolver) (rtt time.Duration
 	}
 	splitEmail := strings.SplitN(address, "@", -1)
 	domain := strings.ToLower(splitEmail[len(splitEmail)-1])
-	result, rtt, err := resolver.LookupHost(domain)
-	if err == nil && len(result) == 0 {
-		err = errEmptyDNSResponse
+	var rtt1, rtt2 time.Duration
+	var resultMX []string
+	var resultA []net.IP
+	resultMX, rtt1, err = resolver.LookupMX(domain)
+	if err == nil && len(resultMX) == 0 {
+		resultA, rtt2, err = resolver.LookupHost(domain)
+		if err == nil && len(resultA) == 0 {
+			err = errEmptyDNSResponse
+		}
 	}
 	if err != nil {
 		problem := dns.ProblemDetailsFromDNSError(err)
 		err = core.MalformedRequestError(problem.Detail)
 	}
+	rtt := rtt1 + rtt2
```

Comment:
FWIW, this later percolates to `VA.DNS.RTT.A` (previously `VA.DNS.RTT.MX`). Not sure if something should be done about that or not.

Also, I think it could all be done in a single DNS query, but the code changes would be more invasive.

I guess this approach is great if it can be granted the _General Availability_ milestone.